mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,552 @@
+
+    def pre_run(self, context: MLClientCtx):
+        """
+        Initializes the MLFlow tracker, setting the experiment name as configured by the user.
+
+        :param context: Current mlrun context
+        """
+        # Check for a user set experiment name via the environment variable:
+        self._experiment_name = (
+            mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.get()
+        )
+
+        # Check if the user configured for matching the experiment name with the context name:
+        if getattr(
+            mlconf.external_platform_tracking, mlflow.__name__
+        ).match_experiment_to_runtime:
+            if self._experiment_name is not None:
+                context.logger.warn(
+                    f"`mlconf.external_platform_tracking.mlflow.match_experiment_to_runtime` is set to True but the "
+                    f"MLFlow experiment name environment variable ('MLFLOW_EXPERIMENT_NAME') is set for using the "
+                    f"name: '{self._experiment_name}'. This name will be overriden "
+                    f"with MLRun's runtime name as set in the MLRun configuration: '{context.name}'."
+                )
+            self._experiment_name = context.name
+            mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.set(
+                self._experiment_name
+            )
+
+        # We have 3 options to track our run, either we set the run id, or we have the experiment, and then we check
+        # for added runs in it, or we do none, and then we look for added runs across all experiments.
+        # In case we want to determine the run id we have to create a new mlflow run, save the run id and use it
+        # to override the run later (due to mlflow limitations)
+        if mlrun.mlconf.external_platform_tracking.mlflow.control_run:
+            with mlflow.start_run() as run:
+                self._run_id = run.info.run_id
+            mlflow.environment_variables.MLFLOW_RUN_ID.set(self._run_id)
+
+        elif self._experiment_name:
+            # We try to load an existing experiment first
+            experiment = mlflow.get_experiment_by_name(name=self._experiment_name)
+            # If no experiment with corresponding name exists, we create a new one
+            if not experiment:
+                experiment_id = mlflow.create_experiment(name=self._experiment_name)
+                experiment = mlflow.get_experiment(experiment_id=experiment_id)
+
+            # Save all runs logged in experiment prior of our run for later comparison
+            self._pre_runs = set(
+                [
+                    run.info.run_id
+                    for run in mlflow.search_runs(
+                        experiment_ids=experiment.experiment_id,
+                        output_format="list",
+                    )
+                ]
+            )
+
+        else:
+            # We warn user that when no experiment name or run id is given we look at all runs and might track the wrong
```

Comment:
I don't see why support this case. Its too much overhead to maintain with little to no value. Users should not use this. IMO lets remove. If you don't want to do it now, please add a TODO to decide / discuss before merging to dev.

Reply:
added TODO